helix-core/src/main/java/org/apache/helix/tools/ClusterVerifiers/ZkHelixClusterVerifier.java

```diff
@@ -187,6 +190,14 @@ public boolean verifyByZkCallback() {
    * @return
    */
   public boolean verifyByPolling(long timeout, long period) {
```

Comment:
I believe you can and should add the same wait for verifyByZkCallback too.
The logic of verifyByZkCallback is that
1. subscribe to changes.
2. no matter if change happens, verify the state for once.
3. if subscription results in some notification, then verify in parallel.
4. any of the paths succeed, then the verify succeeds.

In this case, we shall wait until the first subscribe to ensure the waitTillVerify config takes effect. This won't cause any problem with verifying logic. Since if the change happens before we subscribe, we are still fine because the first default verify check will be done once at least.

Reply:
make sense.